models/data.model.js

```diff
@@ -28,22 +28,57 @@ exports.fetchArticlesById = (id) => {
     });
 };
 
-exports.fetchArticles = () => {
-  return db
-    .query(
-      `SELECT articles.article_id, articles.author, articles.title, articles.topic, articles.created_at, articles.votes, articles.article_img_url, CAST(COUNT(comments.article_id)AS INT) AS comment_count
-      FROM comments
-      RIGHT JOIN articles
-      ON articles.article_id = comments.article_id
-      GROUP BY articles.article_id, articles.author, articles.title, articles.topic, articles.created_at, articles.votes, articles.article_img_url
-      ORDER BY created_at DESC;`
-    )
-    .then((result) => {
-      if (result.rows.length > 0) {
-        const articles = result.rows;
-        return articles;
-      }
-    });
+exports.fetchArticles = (sort_by, order, topic) => {
+  if (
+    sort_by &&
+    sort_by !== "article_id" &&
+    sort_by !== "title" &&
+    sort_by !== "author" &&
+    sort_by !== "comment_count" &&
+    sort_by !== "topic" &&
+    sort_by !== "created_at" &&
+    sort_by !== "article_img_url" &&
+    sort_by !== "votes" &&
+    sort_by !== "body"
+  ) {
+    return Promise.reject({ status: 400, msg: "Invalid Sort Query" });
+  }
+  if (
+    order &&
+    order !== "desc" &&
+    order !== "asc" &&
+    order !== "DESC" &&
+    order !== "ASC"
+  ) {
+    return Promise.reject({ status: 400, msg: "Invalid Order Query" });
+  }
+
+  const queryParams = [];
+  let articlesQueryStr = `SELECT articles.article_id, articles.title, articles.author, articles.topic, articles.created_at, articles.votes, articles.article_img_url, CAST(COUNT(comments.article_id)AS INT) AS comment_count
+  FROM comments
+  RIGHT JOIN articles
+  ON articles.article_id = comments.article_id`;
+
+  if (topic) {
+    articlesQueryStr += ` WHERE topic = '${topic}'`;
```

Comment:
This needs a refactor - at the moment anyone can add a string in here and sql inject malicious code. 

You need to sanitise the topic before adding it to your query string. 

Check the db for the current topics, use that to check if what they've added as a topic is valid, otherwise - reject

Reply:
Dont hardcode the topics - you'll have a horrible shock when it comes to frontend and try to link these up